packages/astro/src/core/sync/index.ts

```diff
@@ -251,7 +251,26 @@ async function syncContentCollections(
 				ssr: { external: [] },
 				logLevel: 'silent',
 			},
-			{ settings, logger, mode, command: 'build', fs, sync: true, routesList, manifest },
+			{
+				// Prevent mutation of specific properties by vite plugins during sync
+				settings: {
+					...settings,
+					// Prevent mutation by vite plugins during sync
+					buildOutput: undefined,
```

Comment:
Do you think we can achieve this using TypeScript? I know it has the type [`Readonly`](https://www.typescriptlang.org/docs/handbook/utility-types.html#readonlytype)